src/Microsoft.Extensions.CodeGeneration.Utils/DotNet/LibraryExportExtensions.cs

```diff
@@ -14,17 +14,28 @@ namespace Microsoft.Extensions.CodeGeneration.DotNet
 {
     public static class LibraryExportExtensions
     {
-        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export)
+        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export, bool throwOnError = true)
         {
             var references = new List<MetadataReference>();
             AssemblyMetadata assemblyMetadata;
             foreach (var lib in export.CompilationAssemblies)
             {
-                using (var stream = File.OpenRead(lib.ResolvedPath))
+                try
                 {
-                    var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
-                    assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
-                    references.Add(assemblyMetadata.GetReference());
+                    using (var stream = File.OpenRead(lib.ResolvedPath))
+                    {
+                        var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
+                        assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
+                        references.Add(assemblyMetadata.GetReference());
+                    }
+                }
+                catch (Exception ex) 
+                {
+                    if(throwOnError) 
+                    {
+                        throw ex;
+                    }
+                    continue;
```

Comment:
What would happen if we continue? we would have missing reference and compilation would fail? How it would look for user? 
Also what kind of exceptions could be here? Can we look for specific ones we expect and let others go through? It is not good in general to catch all exceptions.